libs/ts-rest/express/src/lib/ts-rest-express.ts

```diff
@@ -117,6 +118,45 @@ const recursivelyApplyExpressRouter = (
   }
 };
 
+const validateRequest = (
+  req: Request,
+  res: Response,
+  schema: AppRouteQuery | AppRouteMutation,
+  options: Options
+) => {
+  const paramsResult = checkZodSchema(req.params, schema.pathParams, {
+    passThroughExtraKeys: true,
+  });
+
+  if (!paramsResult.success) {
+    return res.status(400).send(paramsResult.error);
+  }
+
+  const headersResult = checkZodSchema(req.headers, schema.headers, {
+    passThroughExtraKeys: true,
```

Comment:
This should be an option IMO. Can we let the developer choose whether or not they want to `passThroughExtraKeys`?

Reply:
Also, don't we want to parse the headers first, before checking the params?

Reply:
In terms of the order, I'm thinking of it in a top-down fashion when observing a request. In my head it goes: URL (path params) -> headers -> query -> body